### ISSUE TITLE
multiple viewports

### ISSUE THREAD

**Issue by smeeteday** [potree/potree#955]
https://threejs.org/examples/webgl_multiple_views.html
As used in
Capture one scene with multiple cameras
I want to display a point cloud in multiple views.
Used as some multiple views for potree1.7 viewer
It seems that coding is included,
It's difficult and I'm not sure.
Does anyone have used multiple viewports?

**Comment by smcavoy12** [comment]
Yeah, but our implementation is lacking. We can't get the controls to work on both sides, also it totally loads one pointcloud first. We're trying another way with two separate objects on the same page, trying to just mirror mouse input accross the right half.

https://pointcloud.ucsd.edu/OH3D/Bagan_Earthquake_Dual.html

If you figure out an improvement I'd love to see it.

`

  | <!DOCTYPE html>
-- | --
  | <html lang="en">
  |  
  | <head>
  | <meta charset="utf-8">
  | <meta name="description" content="">
  | <meta name="author" content="">
  | <meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=no">
  | <title>Potree Viewer</title>
  |  
  | <link rel="stylesheet" type="text/css" href="../build/potree/potree.css">
  | <link rel="stylesheet" type="text/css" href="../libs/jquery-ui/jquery-ui.min.css">
  | <link rel="stylesheet" type="text/css" href="../libs/openlayers3/ol.css">
  | <link rel="stylesheet" type="text/css" href="../libs/spectrum/spectrum.css">
  | <link rel="stylesheet" type="text/css" href="../libs/jstree/themes/mixed/style.css">
  | </head>
  |  
  | <body>
  | <script src="../libs/jquery/jquery-3.1.1.min.js"></script>
  | <script src="../libs/spectrum/spectrum.js"></script>
  | <script src="../libs/jquery-ui/jquery-ui.min.js"></script>
  | <script src="../libs/three.js/build/three.min.js"></script>
  | <script src="../libs/three.js/extra/lines.js"></script>
  | <script src="../libs/other/BinaryHeap.js"></script>
  | <script src="../libs/tween/tween.min.js"></script>
  | <script src="../libs/d3/d3.js"></script>
  | <script src="../libs/proj4/proj4.js"></script>
  | <script src="../libs/openlayers3/ol.js"></script>
  | <script src="../libs/i18next/i18next.js"></script>
  | <script src="../libs/jstree/jstree.js"></script>
  | <script src="../build/potree/potree.js"></script>
  | <script src="../libs/plasio/js/laslaz.js"></script>
  |  
  |  
  | 
  |  
  | <div class="potree_container" style="position: absolute; width: 100%; height: 100%; left: 0px; top: 0px; display: flex; ">
  | <div style="width: 50%" id="potree_render_area"></div>
  |  
  | </div>
  | <div class="potree_container" style="position: absolute; width: 100%; height: 100%; left: 50%; top: 0px; display: flex; ">
  | <div style="width: 50%" id="potree_render_area2"></div>
  |  
  | </div>
  |  
  |  
  | <script>
  |  
  | window.viewer = new Potree.Viewer(document.getElementById("potree_render_area"));
  |  
  | viewer.setEDLEnabled(true);
  | viewer.setFOV(60);
  | viewer.setEDLStrength(1.3);
  | viewer.setEDLRadius(1.6);
  | viewer.setPointBudget(5 * 1000 * 1000);
  | document.title = "";
  | viewer.setBackground("gradient");// ["skybox", "gradient", "black", "white"];
  | viewer.setDescription(`Bagan- Eim Ya Kyaung, before (right) and after (left) 2016 earthquake`);
  | viewer.loadSettingsFromURL();
  |  
  | viewer.loadGUI(() => {
  | viewer.setLanguage('en');
  | $("#menu_appearance").next().show();
  | $("#menu_tools").next().show();
  | $("#menu_scene").next().show();
  | // viewer.toggleSidebar();
  | });
  |  
  | Potree.loadPointCloud("https://pointcloud.ucsd.edu/OH3D/pointclouds/mn3w-xc18/mn3w-xc18_lidar_terrestrial/metadata.json", "Bagan Pre earthquake mn3w-xc18", e => {
  | let pointcloud = e.pointcloud;
  | let material = pointcloud.material;
  |  
  | material.activeAttributeName = "color";
  | //pointcloud.material.intensityRange = [0,65535];
  | material.minSize = 2;
  | material.pointSizeType = Potree.PointSizeType.ADAPTIVE;
  | material.shape = Potree.PointShape.CIRCLE;
  | e.pointcloud.translateY(11.18);
  | e.pointcloud.translateX(-21);
  | e.pointcloud.translateZ(0.35);
  | e.pointcloud.rotation.set(0,0,-0.045);
  | viewer.scene.addPointCloud(pointcloud);
  | //viewer.fitToScreen();
  |  
  | viewer.scene.view.position.set(-33.575, 19.856, 17.944);
  | viewer.scene.view.lookAt(new THREE.Vector3(-12.761, 0.392, 8.803));
  |  
  | });
  |  
  | window.viewer2 = new Potree.Viewer(document.getElementById("potree_render_area2"));
  |  
  | viewer2.toggleSidebar = () => {
  | // console.log("test");
  | let renderArea = $('#potree_render_area2');
  | let isVisible = renderArea.css('left') !== '50%';
  |  
  | if (isVisible) {
  | renderArea.css('left', '50%');
  | } else {
  | renderArea.css('left', '75%');
  | }
  |  
  | viewer.toggleSidebar();
  | };
  |  
  | viewer2.setEDLEnabled(true);
  | viewer2.setEDLStrength(1.3);
  | viewer2.setEDLRadius(1.6);
  | viewer2.setFOV(60);
  | viewer2.setPointBudget(10 * 1000 * 1000);
  | viewer2.setBackground("gradient");// ["skybox", "gradient", "black", "white"];
  | viewer2.setDescription(`Bagan- Eim Ya Kyaung<br> before 2016 earthquake (left)<a href="https://openheritage3d.org/project.php?id=mn3w-xc18" target=_blank">DOI:10.26301/gsx5-jf85</a> <br> after earthquake (right)<a href="https://openheritage3d.org/project.php?id=gsx5-jf85
  | " target=_blank">DOI:10.26301/mn3w-xc18</a>`);
  | viewer2.loadSettingsFromURL();
  |  
  | viewer2.loadGUI(() => {
  | viewer2.setLanguage('en');
  | });
  |  
  | Potree.loadPointCloud("https://pointcloud.ucsd.edu/OH3D/pointclouds/gsx5-jf85/gsx5-jf85_lidar_terrestrial/metadata.json", "Post Earthquake gsx5-jf85", e => {
  | let pointcloud = e.pointcloud;
  | let material = pointcloud.material;
  |  
  | material.activeAttributeName = "color";
  | //pointcloud.material.intensityRange = [0,65535];
  | material.minSize = 2;
  | material.pointSizeType = Potree.PointSizeType.ADAPTIVE;
  | material.shape = Potree.PointShape.CIRCLE;
  |  
  | viewer2.scene.addPointCloud(pointcloud);
  | //viewer2.fitToScreen();
  | });
  |  
  | // Listen for camera changes
  | viewer.addEventListener('camera_changed', (event) => {
  | // When camera changes, override camera of other viewer
  | viewer2.scene.overrideCamera = event.camera;
  | // Somehow it took me 2 hours to find this `overrideCamera` feature
  | // Hurray for good documentation
  | // viewer2.scene.overrideCamera = null;
  | });
  |  
  | </script>
  | <div style="position: absolute; z-index: 10000; right: 10px; top: 10px;"
  | onclick="window.location.href = '', '_blank'">
  | <img src="/resources/differ.png" />
  | </div>
  | </body>
  |  
  | </html>

`

**Comment by smeeteday** [comment]
Thank you for your fast response. I will try it.

**Comment by smeeteday** [comment]
It is very good. Thank you.
![2021-01-19 16 38 46 d whizimage org f1d57aa8e0fc](https://user-images.githubusercontent.com/329212/105002769-21d1c080-5a75-11eb-84e6-95871c0e8a49.png)

**Comment by yeyan00** [comment]
#1014, multiviews demo,but pointcloud just load once
